sdk/core/azure-core/test/ut/request_id_policy.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+
+#include <gtest/gtest.h>
+
+using namespace Azure::Core;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
+
+namespace {
+class NoOpPolicy : public HttpPolicy {
+public:
+  std::unique_ptr<HttpPolicy> Clone() const override { return std::make_unique<NoOpPolicy>(*this); }
+
+  std::unique_ptr<RawResponse> Send(Request&, NextHttpPolicy, Azure::Core::Context const&)
+      const override
+  {
+    return nullptr;
+  }
+};
+
+constexpr char const* RequestIdHeaderName = "x-ms-client-request-id";
+} // namespace
+
+TEST(RequestIdPolicy, Basic)
+{
+  Request request(HttpMethod::Get, Url("https://www.microsoft.com"));
+
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policies;
+
+    policies.emplace_back(std::make_unique<RequestIdPolicy>());
+    policies.emplace_back(std::make_unique<NoOpPolicy>());
+
+    Azure::Core::Http::_internal::HttpPipeline(policies).Send(request, Azure::Core::Context());
```

Comment:
I am thinking we must have tests where we use all the default policies so we know how they work together.
Right now, we are testing policies individually, which is fine, but we won't catch issues of one policy corrupting or changing another policy's data.

Reply:
Let us add that separately - it might be good to test the default policies stack, plus test the custom policies ("PerRetry/PerRequest")

Reply:
#2027